v-next/hardhat-keystore/src/internal/tasks/delete.ts

```diff
@@ -0,0 +1,49 @@
+import type { KeystoreLoader } from "../types.js";
+import type { UserInteractions } from "../ui/user-interactions.js";
+import type { HardhatRuntimeEnvironment } from "@ignored/hardhat-vnext/types/hre";
+import type { NewTaskActionFunction } from "@ignored/hardhat-vnext/types/tasks";
+
+import { checkMissingKeyTaskArgument } from "../utils/check-missing-key-task-argument.js";
+import { setupRawInterruptionsAndKeystoreLoader } from "../utils/setup-raw-interruptions-and-keystore-loader.js";
+
+interface TaskDeleteArguments {
+  key: string;
+}
+
+const taskDelete: NewTaskActionFunction<TaskDeleteArguments> = async (
+  { key },
+  hre: HardhatRuntimeEnvironment,
+): Promise<void> => {
+  const { keystoreLoader, interruptions } =
+    await setupRawInterruptionsAndKeystoreLoader(hre);
+
+  await remove({ key }, keystoreLoader, interruptions);
+};
+
+export const remove = async (
+  { key }: TaskDeleteArguments,
+  keystoreLoader: KeystoreLoader,
+  interruptions: UserInteractions,
+): Promise<void> => {
+  checkMissingKeyTaskArgument(key, "keystore delete");
```

Comment:
I think that this checks (also for the other tasks) is not needed. There is a default check to ensure that the key is passed